### PR TITLE
metrics fix and refactoring

### PR DESCRIPTION
**Pull request by AlexanderZagaynov**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1671580

### CODE REVIEW

**Comment by bronaghs** [comment]
@AlexanderZagaynov  What makes this a WIP, what is left to be done?

File: app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb [hunk 1]
```diff
@@ -1,165 +1,176 @@
...
+    ## Disk
     {
-      :native_counters       => MEMORY_METERS,
-      :calculation           => ->(stat) { stat.first },
-      :vim_style_counter_key => "mem_usage_absolute_average",
+      :counter_key => 'disk_usage_absolute_average',
+      :source      => :api,
+      :metrics     => [
+        ['Per Disk Read Bytes/sec', 'Per Disk Write Bytes/sec'].freeze, # Microsoft.Compute/virtualMachines
+        ['Disk Read Bytes/Sec', 'Disk Write Bytes/Sec'].freeze, # Microsoft.ClassicCompute/virtualMachines
```
Comment (djberg96) [inline app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb]: We should not care about classicCompute at all.

File: app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb [hunk 2]
```diff
@@ -1,165 +1,176 @@
...
-      :native_counters       => DISK_METERS,
-      :calculation           => ->(stat) { stat.sum / 1.kilobyte },
-      :vim_style_counter_key => "disk_usage_rate_average",
-    }
-  ].freeze
+      :counter_key => 'net_usage_absolute_average',
+      :source      => :api,
+      :metrics     => [
+        ['Network In Total', 'Network Out Total'].freeze, ## Microsoft.Compute/virtualMachines
+        ['Network In', 'Network Out'].freeze, ## Microsoft.ClassicCompute/virtualMachines
```
Comment (djberg96) [inline app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb]: Same.

Comment (AlexanderZagaynov) [reply app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb]: thanks!

File: app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb (hunk 1 above)
Comment (AlexanderZagaynov) [reply app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb]: great! :)

File: app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb [hunk 3]
```diff
@@ -1,231 +1,225 @@
...
-    },
-    "net_usage_rate_average"     => {
-      :counter_key           => "net_usage_rate_average",
-      :instance              => "",
-      :capture_interval      => "20",
-      :precision             => 1,
-      :rollup                => "average",
-      :unit_key              => "kilobytespersecond",
-      :capture_interval_name => "realtime"
+      :counter_key => 'net_usage_absolute_average',
```
Comment (djberg96) [inline app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb]: As per our gitter discussion I think it has to be `net_usage_rate_average` to match the column in the `metrics` table. Same for disk usage.

Comment (AlexanderZagaynov) [reply app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb]: done

**Comment by djberg96** [comment]
Alright, going to merge with the understanding that the specs will be updated along with the memory fix in an upcoming PR.

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 46d1cb3a2f14afb922e6a16cf582a7479e62f5d9
Author: Daniel Berger <dberger@redhat.com>
Date:   Fri Jun 21 13:27:04 2019 -0400

    Merge pull request #335 from AlexanderZagaynov/BZ-1671580_azure_metrics
    
    metrics fix and refactoring
    
    (cherry picked from commit 42764d2bc63e069de4928569f2e9916b16bb8fbb)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1722937
```

**Pull request merged**